internal/provider/utils/utils.go

```diff
@@ -9,3 +14,38 @@ func SetStringValue(input string) types.String {
 	}
 	return types.StringNull()
 }
+
+// Helper function to map the properties response from API to Terraform state
+func HandlePropertiesMapping(ctx context.Context, propertiesAttr types.Map, apiProperties map[string]interface{}) types.Map {
+	// Map API properties to Terraform format
+	properties := map[string]attr.Value{}
+	for key, value := range apiProperties {
+		if value != nil {
+			properties[key] = types.StringValue(value.(string))
+		}
+	}
+
+	// Retrieve the current Terraform state for `properties`
+	var currentPropertiesState map[string]string
+	diags := propertiesAttr.ElementsAs(ctx, &currentPropertiesState, false)
+	if diags.HasError() {
+		tflog.Error(ctx, "Error reading current properties state", map[string]interface{}{"errors": diags.Errors()})
+	}
+
+	// API response is empty
```

Comment:
```suggestion
	// API response has no properties
```

Reply:
No, `API response has no properties` can be interpreted as properties: null or properties: {}. There is important distinction here between these two, API response is empty means more clearly that properties: {}

Reply:
I see, but I still think we can have a better comment here since it is not the API response that is empty rather it is the API response properties, right?